src/repositories/component/component-complex.repository.ts

```
@@ -0,0 +1,186 @@
+   */
+  async upsertComponentWithRelationships(component: {
+    repository: string;
+    branch?: string;
+    id: string;
+    name: string;
+    kind: string;
+    status: ComponentStatus;
+    depends_on?: string[] | null;
+  }): Promise<Component | null> {
+    const repositoryNodeId = String(component.repository);
+    const logicalRepositoryName = this.validateRepositoryNodeId(repositoryNodeId, 'upsertComponentWithRelationships');
```

ESLint

[error] Replace `repositoryNodeId,·'upsertComponentWithRelationships'` with `⏎······repositoryNodeId,⏎······'upsertComponentWithRelationships',⏎····`

(prettier/prettier)